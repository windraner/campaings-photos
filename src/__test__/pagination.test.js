import React from 'react';
import Pagination from '../components/main_container/bottom_bar/pagination/pagination';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const mock = {
  page: 1,
  pagesCount: 10,
  stateHandler() {},
};

describe('Pagination', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Pagination {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
