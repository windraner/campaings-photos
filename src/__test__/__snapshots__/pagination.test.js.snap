// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination matches the snapshot 1`] = `
<div
  className="d-flex"
>
  <div
    className="campaigns-photos__pagination-button campaigns-photos__pagination-button-prev position-relative campaigns-photos__pagination-button__disabled"
    onClick={null}
  >
    <span
      className="campaigns-photos__pagination-left-icon"
    />
    Prev
  </div>
  <div
    className="campaigns-photos__pagination-page campaigns-photos__pagination-page__active"
    key="0"
    onClick={null}
  >
    1
  </div>
  <div
    className="campaigns-photos__pagination-page"
    key="1"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="campaigns-photos__pagination-page"
    key="2"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="campaigns-photos__pagination-page"
    key="3"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="campaigns-photos__pagination-page"
    key="4"
    onClick={[Function]}
  >
    5
  </div>
  <div
    className="campaigns-photos__pagination-separator"
    key="5"
  >
    ...
  </div>
  <div
    className="campaigns-photos__pagination-page"
    key="6"
    onClick={[Function]}
  >
    10
  </div>
  <div
    className="campaigns-photos__pagination-button campaigns-photos__pagination-button-next position-relative"
    onClick={[Function]}
  >
    Next
    <span
      className="campaigns-photos__pagination-right-icon"
    />
  </div>
</div>
`;
